### PR TITLE
User Activation & Password Reset

### PR DESCRIPTION
**Pull request by syahidazuri**

### CODE REVIEW

**Comment by syahidazuri** [comment]
Ready

File: app/assets/javascripts/account_activations.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (manhnt-0960) [inline app/assets/javascripts/account_activations.js]: please delete this empty file

File: app/controllers/account_activations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class AccountActivationsController < ApplicationController
+  def edit
+      user = User.find_by email: params[:email]
```
Comment (manhnt-0960) [inline app/controllers/account_activations_controller.rb]: check indent again

File: app/models/user.rb [hunk 3]
```diff
@@ -36,9 +39,37 @@ def forget
...
+
+  def send_password_reset_email
+    UserMailer.password_reset(self).deliver_now
+  end
+
+  def password_reset_expired?
+    reset_sent_at < 2.hours.ago
+  end
+
+
```
Comment (manhnt-0960) [inline app/models/user.rb]: remove empty line

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
-  get 'sessions/new'
+  get "password_resets/new"
+  get "password_resets/edit"
+  get "sessions/new"
```
Comment (manhnt-0960) [inline config/routes.rb]: remove those line

**Comment by syahidazuri** [comment]
Ready

**Pull request merged**